File: tenspiler/dexter/cpp/for_synthesis/darken_blend_8.cc
#include <vector>
using namespace std;

vector<vector<int>> darken_blend_8(vector<vector<int>> base, vector<vector<int>> active)
{
    vector<vector<int>> out;
    int m = base.size();
    int n = base[0].size();
	for (int row = 0; row < m; row++) {
        vector<int> row_vec;
		for (int col = 0; col < n; col++) {
			int pixel;
			if (base[row][col] > active[row][col])
				pixel = active[row][col];
			else
				pixel = base[row][col];
			row_vec.push_back(pixel);
		}
		out.push_back(row_vec);
	}
	return out;
}

File: tests/llvm/tenspiler/benchmark/numba/darken_blend_8.py
from numba import njit


@njit
def darken_blend_8(base, active):
    output = []
    m = len(base)
    n = len(base[0])
    for i in range(m):
        curr_row = []
        for j in range(n):
            if base[i][j] > active[i][j]:
                pixel = active[i][j]
            else:
                pixel = base[i][j]
            curr_row.append(pixel)
        output.append(curr_row)
    return output
